lib/Backend/BackwardPass.cpp

```diff
@@ -2735,6 +2848,168 @@ BackwardPass::ProcessBlock(BasicBlock * block)
 #endif
 }
 
+bool 
+BackwardPass::CanDeadStoreInstrForScopeObjRemoval(Sym *sym) const
+{
+    if (tag == Js::DeadStorePhase && this->currentInstr->IsStackArgsEnabled())
+    {
+        Func * currFunc = this->currentInstr->m_func;
+        switch (this->currentInstr->m_opcode)
+        {
+            case Js::OpCode::LdHeapArguments:
+            case Js::OpCode::LdHeapArgsCached:
+            case Js::OpCode::LdLetHeapArguments:
+            case Js::OpCode::LdLetHeapArgsCached:
+            {
```

Comment:
Sure. Done. Added a detailed comment

---
In reply to: [68150708](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r68150708) [](ancestors = 68150708)